packages/contract/src/test-utils/test-node-launcher.test.ts

```diff
@@ -0,0 +1,115 @@
+import { FuelError } from '@fuel-ts/errors';
+import { expectToThrowFuelError, safeExec } from '@fuel-ts/errors/test-utils';
+import { Provider } from '@fuel-ts/providers';
+import { WalletConfig } from '@fuel-ts/wallet/test-utils';
+import { join } from 'path';
+
+import { TestNodeLauncher } from './test-node-launcher';
+
+const pathToContractRootDir = join(__dirname, '../../test/fixtures/simple-contract');
+
+describe('TestNodeLauncher', () => {
+  test('kills the node after going out of scope', async () => {
+    let url = '';
+
+    {
+      await using launched = await TestNodeLauncher.launch();
+
+      const { provider } = launched;
+
+      url = provider.url;
+      await provider.getBlockNumber();
+    }
+
+    const { error } = await safeExec(async () => {
+      const p = await Provider.create(url);
+      await p.getBlockNumber();
+    });
+
+    expect(error).toMatchObject({
+      code: 'ECONNREFUSED',
+    });
+  });
+
+  test('kills the node if error happens post-launch', async () => {
+    const port = '9876';
+    try {
+      await TestNodeLauncher.launch({
+        nodeOptions: { port },
+        deployContracts: [{ contractDir: 'invalid path' }],
+      });
+    } catch (err) {
+      expect(err).toBeDefined();
+
+      const { error } = await safeExec(async () => {
+        const url = `http://127.0.0.1:${port}/graphql`;
+        const p = await Provider.create(url);
+        await p.getBlockNumber();
+      });
+
+      expect(error).toMatchObject({
+        code: 'ECONNREFUSED',
+      });
+
+      return;
+    }
+
+    // Should never reach here; using fail() gives a ReferenceError and would crash the whole program
+    expect(false).toBe(true);
+  });
+
+  test('a contract can be deployed', async () => {
+    // #region TestNodeLauncher-deploy-contract
+    await using launched = await TestNodeLauncher.launch({
+      deployContracts: [{ contractDir: pathToContractRootDir }],
+    });
+
+    const {
+      contracts: [contract],
+    } = launched;
```

Comment:
Unfortunately, destructuring of disposables [isn't supported](https://github.com/microsoft/TypeScript/issues/55527) so it's a two-step process.